### ISSUE TITLE
Allow default values for python templater

### ISSUE THREAD

**Issue by snkekorfus** [sqlfluff/sqlfluff#6595]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Current Behavior:
When using f-strings to insert variables in SQL code, we are required to define each variable in the [sqlfluff:templater:python:context] section of the .sqlfluff file. However, in many cases, the actual value of the variable is irrelevant, leading to unnecessary clutter in the .sqlfluff file with multiple template entries.

Proposed Enhancement:
It would be helpful to have the ability to define a default template value for variables when no specific value is provided. This would reduce the need to manually define each variable in the context, streamlining the file and reducing maintenance overhead.

### Current Scenario:
Suppose you have an SQL query that dynamically inserts variables using f-strings in Python, like so:

```python
query = f"""
    SELECT * FROM users
    WHERE age > {min_age}
    AND city = '{city_name}';
"""
```

To make this query pass SQLFluff checks, you currently need to define min_age and city_name in the [sqlfluff:templater:python:context] section of your .sqlfluff file, even if their specific values are irrelevant to the linting process:

```
[sqlfluff:templater:python:context]
min_age = 18
city_name = 'New York'
```

This requirement results in numerous variables being added to the .sqlfluff file, which can become cumbersome, especially in larger projects where many such f-string variables are used.

### Proposed Scenario (With Default Template Values):
With the enhancement, you would no longer need to explicitly define every variable unless necessary. SQLFluff would use a default template value for variables without a specific context entry. For example:

```
[sqlfluff:templater:python:context]
sqlfluff.default = "Placeholder"
min_age = 18  # Explicitly defined, as needed
```

In this case, since city_name is not critical for the linting, SQLFluff would automatically assign the default template value from sqlfluff.default ("Placeholder"), reducing the need for additional entries in the context file.

This simplifies the .sqlfluff file and makes it easier to maintain, especially when dealing with many dynamic variables that don’t require specific values during linting.

Further, we could also remove the min_age and just set sqlfluff.default to any number as it is not important what placeholder is inserted into the sql for min_age and city_name for the linting to work correctly.

### Dialect

ANSI
Databricks

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
